std/sumtype.d

```diff
@@ -1466,6 +1485,30 @@ version (D_BetterC) {} else
     MySum y; y = b;
 }
 
+// Type index
+@safe unittest
+{
+    alias MySum = SumType!(int, float);
+
+    assert(MySum(42).typeIndex == IndexOf!(int, MySum.Types));
+    assert(MySum(3.14).typeIndex == IndexOf!(float, MySum.Types));
+}
+
+// Type index for qualified SumTypes
+// Disabled in BetterC due to use of dynamic arrays
+version (D_BetterC) {} else
+@safe unittest
+{
+    alias MySum = SumType!(const(int[]), int[]);
+
+    int[] ma = [1, 2, 3];
```

Comment:
```suggestion
    int[3] ma = [1, 2, 3];
```

Wouldn't that be enough to make it work in `betterC` ?

Reply:
In theory, yes. In practice, if I use a stack-allocated static array and construct the `SumType` with a slice (i.e., `MySum(ma[])`), the tests will fail even with `-preview=dip1000`, because the workaround for [issue 21229][1] used in `SumType`'s constructors does not correctly preserve lifetime information about the constructor's argument.

[1]: https://issues.dlang.org/show_bug.cgi?id=21229